### ISSUE TITLE
Jquery and Boostrap previous release configuration issue

### ISSUE THREAD

**Issue by waqas-mehmood-pk** [JeffreyWay/laravel-mix#842]
- Laravel Mix Version: 0.12.1 
- Node Version : 6.9.1
- NPM Version :  3.10.8
- OS: Windows

### Description:
Hello Experts
I am a newbie to use web-pack and laravel-mix. I have an old project and with old javascript libraries... 
If I use default laravel-mix setting than jquery not loaded and when I add these both libraries manually in the laravel.mix.js file than bootstrap not working... 

I have fought since last week but no success. 

### Steps To Reproduce:
I have uploaded an example in github repository. 
link: https://github.com/whfamily2006/laravel-mix-jquery-bootstrap

**Comment by ankurk91** [comment]
You are using ``scripts()`` and ``styles()`` methods, it means you are not using webpack. 
Default laravel-mix config that comes with laravel won't work for you for sure.

What do you meant by bootstrap not working ? Any error ?

Did you include all of your build files in [welcome.blade.php](https://github.com/whfamily2006/laravel-mix-jquery-bootstrap/blob/master/resources/views/welcome.blade.php) correctly and in-order ?

**Comment by whfamily2006** [comment]
Thank @ankurk91 for the response. please view the javascript console for error. 
I have used the old version of jquery and bootstrap so that's why I can't use the default laravel-mix settings. so that's why I was creating an example repository if anyone wants to review.

**Comment by ankurk91** [comment]
Your order of js files in ``webpack.mix.js`` is wrong.
You should include jquery before bootstrap.

So here is the order -
1. jQuery (because bootstrap requires jquery to be loaded first)
2. Bootstrap
3. jQuery plugins
4. Your custom code - main.js

Also you have included bootstrap css twice.

**Comment by whfamily2006** [comment]
Here is my webpack.mix.js file

```
const RESOURCE_PATH = `resources/assets/custom`;
const NODE_MODULES_PATH = `node_modules`;

mix
    .scripts(
        [
            **`${RESOURCE_PATH}/js/jquery-2.0.2.min.js`**,  
            **`${NODE_MODULES_PATH}/bootstrap/dist/js/bootstrap.js`**, 
            `${RESOURCE_PATH}/js/jarvis.widget.min.js`,
            `${RESOURCE_PATH}/js/smoothscroll.js`,
            `${RESOURCE_PATH}/js/mousescroll.js`,
            `${RESOURCE_PATH}/js/jquery-ui-1.10.3.min.js`,
            `${RESOURCE_PATH}/js/HoldOn.min.js`,
            `${RESOURCE_PATH}/js/jquery.inview.min.js`,
            `${RESOURCE_PATH}/js/jquery.isotope.min.js`,
            `${RESOURCE_PATH}/js/jsUKpostcode.js`,
        ], `public/js/app.js`)

    .scripts(
        [
            `${NODE_MODULES_PATH}/bootstrap-select/dist/js/bootstrap-select.js`,
            `${RESOURCE_PATH}/js/SmartNotification.min.js`,
            `${NODE_MODULES_PATH}/sweetalert/dist/sweetalert.min.js`,
            `${RESOURCE_PATH}/js/pace.min.js`,
            `${NODE_MODULES_PATH}/owl.carousel/dist/owl.carousel.min.js`,
            `${NODE_MODULES_PATH}/wowjs/dist/wow.min.js`,
            `${RESOURCE_PATH}/js/jquery.prettyPhoto.js`,
            `${RESOURCE_PATH}/js/datatables.bootstrap.js`,
            **`${RESOURCE_PATH}/js/main.js`**,
        ], `public/js/packages.js`)
    .styles(
        [
           `${NODE_MODULES_PATH}/bootstrap/dist/css/bootstrap.css`, 
            `${RESOURCE_PATH}/css/main.css`,
            `${RESOURCE_PATH}/css/flag-icon.css`,
            `${RESOURCE_PATH}/css/owl.transitions.css`,
            `${RESOURCE_PATH}/css/segmented-control.css`
        ], `public/css/custom.css`)

    .styles(
        [
            `${NODE_MODULES_PATH}/bootstrap/dist/css/bootstrap.min.css`,
            `${NODE_MODULES_PATH}/animate.css/animate.min.css`,
            `${NODE_MODULES_PATH}/owl.carousel/dist/assets/owl.carousel.css`,
            `${NODE_MODULES_PATH}/hover.css/css/hover.css`,
            `${NODE_MODULES_PATH}/bootstrap-select/dist/css/bootstrap-select.css`,
            `${NODE_MODULES_PATH}/sweetalert/dist/sweetalert.css`,
            `${NODE_MODULES_PATH}/sweetalert/themes/twitter/twitter.css`
        ], `public/css/vendor.css`)

    .sourceMaps();
```
@ankurk91 I have changed according to your suggestions.
1. Jquery already at the top of the first scripts function. 
2. Bootstrap added now after jquery line in first scripts function too.
3. Jquery plugins
4. my custom code main.js file at the end of 2nd scripts function.
5. and in the last but not the least before now I have not included the bootstrap file but now I added and others bootstrap files are included for select2, and datatable package.

but the error is still there. 
##`TypeError: false is not a function`, 

** please clone the repo and have a look, I need your kind help.**

**Comment by ruchern** [comment]
Your `webpack.mix.js` file is too messy.

You should be having `app.js` to handle this:

For example: 
```javascript
window.$ = window.jQuery = require('jquery')
```

Then in your `webpack.mix.js` file do this:

```javascript
mix.scripts('resources/js/app.js, 'app.js')
```

**Comment by whfamily2006** [comment]
@ankurk91 thank you for an explanation, I know this is not the relevant forum for my silly questions, but I search these since last one and half week and don't understand still.

Please also explain these following: 
1. window and window.select2 (select2 is package but what is `window` and what is `window.select2`) if we write `window.select2` that's mean select2.js and if select2.css exists then it's included automatically, if not included automatically then we add it in app.scss?
2. Please also explain `window.$` why we need to `.$` ?
3. If I have Jquery old version mean not npm supported version than how can I include that version
is this work `window.$ = window.jQuery = require('./../custom/js/jquery-2.0.2.min')` ?

**Comment by ankurk91** [comment]
[webpack](https://laracasts.com/index/webpack) tutorials, 
[JS](https://www.codeschool.com/learn/javascript) tutorials

**Comment by ruchern** [comment]
@whfamily2006 In layman terms, `window` is basically global. So you can use `select()` at any pages of your code.

As @ankurk91 mentioned, yeah you should watch some of the tutorials on Webpack. Laracasts is a good place to start. https://laracasts.com/series/webpack-for-everyone

**Comment by whfamily2006** [comment]
@ruchern thank you for explanation. just last thing if I have old version of jquery (jquery.js file downloaded) which not installed via npm so how can I include that download jquery js file in app.js `window.$ = window.jQuery = require('jquery')`

I have seen webpack-for-everyone, laravel-mix on laracast and then I was trying to configure.

**Comment by ruchern** [comment]
In your `webpack.mix.js`

```javascript
mix.scripts('path/to/jquery.min.js', 'public/js')
```

**Comment by ankurk91** [comment]
You can install old version through npm like this -
```
npm install jquery@1.12.4 --save
```

**Comment by whfamily2006** [comment]
Thank you @ruchern, @ankurk91 for your time and tolerating me. 
@ankurk91 I have used `jQuery v2.0.2` and npm not support this version. 

![image](https://cloud.githubusercontent.com/assets/9881496/26668653/2980c5a6-46c4-11e7-93e4-28f5981bd4a8.png)

**Comment by ankurk91** [comment]
seems like they have not published that version on npm.
You can install colser version.

npm has these jQuery version available-
```
npm ERR! notarget 3.2.1, 3.2.0, 3.1.1, 3.1.0, 3.0.0, 3.0.0-rc1, 3.0.0-beta1, 3.0.0-alpha1, 2.2.4, 2.2.3, 2.2.2, 2.2.1, 2.2.0, 2.1.4, 2.1.3, 2.1.2, 2.1.1, 2.1.1-rc2, 2.1.1-rc1, 2.1.1-beta1, 2.1.0, 2.1.0-rc1, 2.1.0-beta3, 2.1.0-beta2, 1.12.4, 1.12.3, 1.12.2, 1.12.1, 1.12.0, 1.11.3, 1.11.2, 1.11.1, 1.11.1-rc2, 1.11.1-rc1, 1.11.1-beta1, 1.11.0, 1.11.0-rc1, 1.11.0-beta3, 1.9.1, 1.8.3, 1.8.2, 1.7.3, 1.7.2, 1.6.3, 1.6.2, 1.5.1
```

**Comment by ruchern** [comment]
@ankurk92 And they are unlikely to publish 2.0.2.

**Comment by endrureza** [comment]
well, it doesn't work totally for me.

first, my 

**webpack.mix.js** 
`
let mix = require('laravel-mix');
mix.js('resources/assets/back/js/app.js','public/js/back.js')
	   .sass('resources/assets/back/scss/app.scss','public/css/back.css')
	   .disableNotifications();
`

**app.js**
`
/**
 * Load Main Component of Javascript
 */
window.$ = window.jQuery = require('jquery');
require('bootstrap-datepicker') // or window.datepicker = require('bootstrap-datepicker')
`

it will cause this error

`
$(...).datepicker is not a function
`

@ankurk91 @ruchern am i making mistake here?

**Comment by ankurk91** [comment]
@endrureza 
Please follow

https://github.com/JeffreyWay/laravel-mix/blob/master/docs/autoloading.md

**Comment by endrureza** [comment]
I've done that before

**webpack.mix.js**

```
mix.autoload({
   jquery: ['$', 'window.jQuery']
});

let mix = require('laravel-mix');

mix.js('resources/assets/back/js/app.js','public/js/back.js')
	   .sass('resources/assets/back/scss/app.scss','public/css/back.css')
	   .disableNotifications();
```

**app.js**

```
window.$ = window.jQuery = require('jquery');

require('bootstrap-datepicker') // or window.datepicker = require('bootstrap-datepicker')
```

it caused this error 

```
$ is not a function
```

@ankurk91 @ruchern i'm getting more confused
